autopilot/agent.go

```diff
@@ -470,23 +520,34 @@ func (a *Agent) controller(startingBalance btcutil.Amount) {
 							directive.ChanAmt, err)
 
 						// As the attempt failed, we'll
-						// clear it from the set of
-						// pending channels.
+						// disconnect the peer and clear
+						// it from the set of pending
+						// channels.
+						err = a.cfg.DisconnectPeer(
```

Comment:
could this be a peer we were already connected to? In that case we probably shouldn't disconnect.

Reply:
Context wise, I guess we could determine by returning a bool from connect which denotes if it's an existing channel?